src/ASM/ASMs2Dmx.C

```diff
@@ -1045,23 +1045,27 @@ bool ASMs2Dmx::evalSolutionPiola (Matrix& sField, const Vector& locSol,
   std::vector<std::vector<Go::BasisPtsSf>> splinex(m_basis.size());
   std::vector<Go::BasisDerivsSf> splineg;
 
-  if (std::any_of(nfx.begin(), nfx.end(),
+  if (std::any_of(nfx.begin(), nfx.begin() + 2,
                   [](const unsigned char in) { return in > 1; })) {
     std::cerr << "*** ASMs2Dmx::evalSolution: Piola mapping requires "
-              << "a single field on each basis" << std::endl;
+              << "a single field on each Piola mapped basis" << std::endl;
     return false;
   }
 
-  size_t len = std::accumulate(nb.begin(),nb.end(),0u);
+  size_t len = std::inner_product(nb.begin(), nb.end(), nfx.begin(), 0u);
   bool withPressure = len == locSol.size();
+  size_t nFields;
   if (!withPressure)
-    len = nb[0] + nb[1];
+    len = nb[0] + nb[1], nFields = 2;
+  else
+    nFields = std::accumulate(nfx.begin(), nfx.end(), 0);
```

Comment:
Maybe better with
```
  size_t nFields = 2;
  if (withPressure)
    nFields = std::accumulate(nfx.begin(), nfx.end(), 0u);
  else
    len = nb[0] + nb[1];
```
The logic is in any case hard to grasp

Reply:
okay, it will cause sca warnings but those are harmless of course.